src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2674,6 +2755,42 @@ theorem getLsbD_intMax (w : Nat) : (intMax w).getLsbD i = decide (i + 1 < w) :=
   · rw [Nat.sub_add_cancel (Nat.two_pow_pos (w - 1)), Nat.two_pow_pred_mod_two_pow (by omega)]
 
 
+/-! ### abs -/
+
+theorem abs_def {x : BitVec w} : x.abs = if x.msb then .neg x else x := rfl
+
+theorem abs_eq_if (x : BitVec w) : x.abs = 
+    if x.msb = true then
+      if x = BitVec.intMin w then (BitVec.intMin w) else -x
+    else x := by
+  · rw [BitVec.abs_def]
+    by_cases hx : x.msb = true <;> by_cases hx' : x = BitVec.intMin w <;> simp [hx, hx']
```

Comment:
It more precisely says what happens at `intMin`, and in the other cases, describes the value in terms of what "one would expect".

I agree that, mathematically, the case of `if x = BitVec.intMin w then (BitVec.intMin w)` is redundant, since we have the equality `- (BitVec.intMin w) = BitVec.intMin w`.

However, this (at least to me) makes thinking about the entire operation much harder.  Psychologically, it feels much cleaner to me to state it as:

- If the number is positive, then it stays unchanged
- If it's negative, and equal to `intMin`, then it stays `intMin`
- otherwise, it's negative and not equal to `intMin`, and is negated to become positive. 


---

I don't have strong opinions on this, but I try to explain why I felt the need to add a new definition :)